src/all-transfers/index.ts

```diff
@@ -142,19 +146,49 @@ async function main() {
     }
 
     for (const tx of res.data.transactions) {
+      //// If we get all value transfers from traces we can remove this.
+      //   if (tx.gasUsed == undefined || tx.gasPrice == undefined) {
+      //     console.log("values are undefined");
+      //     continue;
+      //   }
+      //
+      //   // transaction_count_out++; // Should we increment this when paying for gas?
+      //   total_wei_volume_out += BigInt(tx.gasUsed) * BigInt(tx.gasPrice);
+
+      if (!tx.from || !tx.to || tx.value == undefined || tx.gasUsed == undefined || tx.gasPrice == undefined) {
+        console.log("values are undefined", tx);
+        continue;
+      }
+
+      if (tx.from === targetAddress) {
+        transaction_count_out++;
+        total_wei_volume_out += BigInt(tx.value) + (BigInt(tx.gasUsed) * BigInt(tx.gasPrice));
+      } /*if (tx.to === targetAddress)*/ else {
+        transaction_count_in++;
+        total_wei_volume_in += BigInt(tx.value);
+      }
+    }
+    for (const tx of res.data.traces) {
       if (!tx.from || !tx.to || tx.value == undefined) {
         console.log("values are undefined");
         continue;
       }
 
+      //// Sanity check
+      //assert(tx.from === targetAddress || tx.to == targetAddress, `Invalid trace data, neither from nor to is the target address, from: ${tx.from}, to: ${tx.to}, target: ${targetAddress}`)
+      // if (tx.from !== targetAddress && tx.to !== targetAddress) {
+      //   continue; // We will be able to improve this when we have better join modes.
+      // }
+
       if (tx.from === targetAddress) {
```

Comment:
The top level transaction might be duplicated in the traces as well, not 100% sure about this but I remember it being like that